src/main/java/ch/njol/skript/lang/Variable.java

```diff
@@ -119,7 +125,26 @@ private Variable(VariableString name, Class<? extends T>[] types, boolean local,
 	 * @return true if the name is valid, false otherwise.
 	 */
 	public static boolean isValidVariableName(String name, boolean allowListVariable, boolean printErrors) {
-		name = name.startsWith(LOCAL_VARIABLE_TOKEN) ? "" + name.substring(LOCAL_VARIABLE_TOKEN.length()).trim() : "" + name.trim();
+		assert !name.isEmpty(): "Variable name should not be empty";
+		char first = name.charAt(0);
+		check_reserved_tokens:
+		for (char token : reservedTokens) {
+			if (first == token && printErrors) {
+				/*
+				A lot of people already use '-' so we want to skip this warning iff they're using it here
+				*/
+				if (first == '-') {
+					for (VariablesStorage store : Variables.getStores()) {
+						@Nullable Pattern pattern = store.getNamePattern();
+						if (pattern != null && pattern.pattern().equals("(?!-).*"))
+							continue check_reserved_tokens;
+					}
+				}
+				Skript.warning("The character '" + token + "' is reserved at the start of variable names, " +
```

Comment:
I would like to see this message in `english.lang`. I think it is reasonable for us to provide translation support for these internal messages.